modules/navier_stokes/src/fvkernels/INSFVMomentumAdvection.C

```diff
@@ -449,13 +448,6 @@ INSFVMomentumAdvection::interpolate(Moose::FV::InterpMethod m,
 
   if (onBoundary(*_face_info))
   {
-    // In my mind there should only be about one bc_id per FaceInfo
-    mooseAssert(_face_info->boundaryIDs().size() == 1,
-                "I think some of my logic might depend on my implicit assumption that we have "
-                "one boundary ID at most per face");
-    mooseAssert(_flow_boundaries.find(*_face_info->boundaryIDs().begin()) != _flow_boundaries.end(),
-                "INSFV*Advection flux kernel objects should only execute on flow boundaries.");
```

Comment:
I don't want this assert to go away. I really want to make sure this kernel only ever executes on a flow boundary if it's executing on a boundary. So you could do this:
```c++
#ifndef NDEBUG
bool flow_boundary_found = false;
for (const auto bid : _face_info->boundaryIDs())
  if (_flow_boundaries.find(bid) != _flow_boundaries.end())
    {
      flow_boundary_found = true;
      break;
    }

mooseAssert(flow_boundary_found, "INSFV*Advection flux kernel objects should only execute on flow boundaries.");
#endif
```

Reply:
Sounds good. I removed it because I didnt feel like looping through all faces on every Qp. But I guess there's no other way to test where a kernel is executed

Reply:
In most cases there will only be one boundary ID to loop over so the complexity most times will be O(log(flow_boundaries.size())). And there should only ever be one qp corresponding to the face centroid. If we have more than one qp we are in trouble. 

With the ifndef guards this won’t be executed in optimized code so most users won’t notice any slow down from this sanity check. I don’t care if dbg or devel runs slowly